File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        i = 0
        idx = 0
        while i < len(chars):
            ch = chars[i]
            count = 0
            while i < len(chars) and chars[i] == ch:
                i += 1
                count += 1
            chars[idx] = ch
            idx += 1
            if count > 1:
                for digit in str(count):
                    chars[idx] = digit
                    idx += 1
        return idx

File: 443-string-compression/string-compression.java
class Solution {
    public int compress(char[] chars) {
        int inx = 0;
        int i = 0;
        int n = chars.length;
        while(i < n){
            int cnt = 0, j = i;
            char ch = chars[i];
            while(j < n && chars[j] == ch){
                cnt++;
                j++;
                i++;
            }
            chars[inx++] = ch;
            if(cnt > 1){
                String s = String.valueOf(cnt);
                int k = 0;
                while(k < s.length()){
                    chars[inx++] = (char)(s.charAt(k++));
                }
            }
        }
        return inx;
    }
}